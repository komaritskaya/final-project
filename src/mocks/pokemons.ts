import { Pokemon } from '../types';

const pokemons: Pokemon[] = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: true,
    catchDate: new Date(),
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
];

export default pokemons;
