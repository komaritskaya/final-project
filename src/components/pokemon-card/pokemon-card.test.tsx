import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import PokemonCard from './pokemon-card';
import { pokemons } from '../../mocks/pokemons';

it('Should Pokemon Card component render correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <PokemonCard
          pokemon={pokemons[0]}
        />
      </Router>,
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
