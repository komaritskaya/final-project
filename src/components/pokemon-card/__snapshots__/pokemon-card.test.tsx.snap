// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Pokemon Card component render correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <a
    href="/1"
    onClick={[Function]}
    style={
      Object {
        "color": "inherit",
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="bulbasaur"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        src="/img/1.png"
        title="bulbasaur"
      />
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          BULBASAUR
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </a>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        catch
      </span>
    </button>
  </div>
</div>
`;
